### PR TITLE
Add list-group feature and some improvements

### PR DESCRIPTION
**Pull request by yasassri**
## Purpose
This PR contains  
- List-group feature and tests.
- ReadMe and sample updates.
- Test improvements

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java [hunk 1]
```diff
@@ -81,14 +82,66 @@ public void runTest(String sourceRoot, Path[] sourceFilePaths, List<String> grou
...
+     * lists the groups available in tests.
+     *
+     * @param sourceRoot      source root of the project
+     * @param sourceFilePaths package or program file paths
+     * @return
+     */
+    public void listGroups(String sourceRoot, Path[] sourceFilePaths) {
+        //Build the test suites
+        compileAndBuildSuites(sourceRoot, sourceFilePaths);
+        List groupList = getGroupList();
```
Comment (nirmal070125) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: Shall we return a List\<String\> here?

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java [hunk 2]
```diff
@@ -81,14 +82,66 @@ public void runTest(String sourceRoot, Path[] sourceFilePaths, List<String> grou
...
+    /**
+     * Returns a distinct list of groups in test functions.
+     *
+     * @return a list of groups
+     */
+    public List getGroupList() {
+
+        Map<String, TestSuite> testSuites = TesterinaRegistry.getInstance().getTestSuites();
+        if (testSuites.isEmpty()) {
+            throw new BallerinaException("No test functions found in the provided ballerina files.");
```
Comment (nirmal070125) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: Suggestion to change the error msg to "No tests found in the Ballerina project."

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java [hunk 3]
```diff
@@ -81,14 +82,66 @@ public void runTest(String sourceRoot, Path[] sourceFilePaths, List<String> grou
...
+        }
+        List<String> groupList = new ArrayList<>();
+        testSuites.forEach((packageName, suite) -> {
+            suite.getTests().forEach(test -> {
+                if (test.getGroups().size() > 0) {
+                    groupList.addAll(test.getGroups());
+                }
+            });
+        });
+        return groupList.stream().distinct().collect(Collectors.toList());
```
Comment (nirmal070125) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: Shall we print a message like "No test groups found in the Ballerina project." when groupList is empty?

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java [hunk 4]
```diff
@@ -165,14 +165,16 @@ public void process(FunctionNode functionNode, List<AnnotationAttachmentNode> an
...
-                                if (attributeNode.getValue() instanceof BLangArrayLiteral) {
-                                    BLangArrayLiteral values = (BLangArrayLiteral) attributeNode.getValue();
-                                    boolean isGroupPresent = isGroupAvailable(groups, values.exprs.stream().map(node
-                                            -> node.toString()).collect(Collectors.toList()));
+
+                        // check if groups attribute is present in the annotation
+                        if (GROUP_ANNOTATION_NAME.equals(name)) {
+                            if (attributeNode.getValue() instanceof BLangArrayLiteral) {
+                                BLangArrayLiteral values = (BLangArrayLiteral) attributeNode.getValue();
+                                test.setGroups(values.exprs.stream().map(node
```
Comment (nirmal070125) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestAnnotationProcessor.java]: this line and below seems to have not formatted.

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestCmd.java [hunk 5]
```diff
@@ -63,6 +63,9 @@
     @Parameter(names = "--ballerina.debug", hidden = true, description = "remote debugging port")
     private String ballerinaDebugPort;
 
+    @Parameter(names = { "--list-groups", "-lg" }, hidden = true)
+    private boolean listGroups;
```
Comment (nirmal070125) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/TestCmd.java]: I think it's better to make it available via the help. hidden=false

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/entity/Test.java [hunk 6]
```diff
@@ -119,4 +120,12 @@ public void addDependsOnTestFunction(String function) {
     public void addDependsOnTestFunction(TesterinaFunction function) {
         this.dependsOnTestFunctionObjs.add(function);
     }
+
+    public void setGroups(List<String> groups) {
+        this.groups = groups;
+    }
+
+    public List getGroups() {
```
Comment (nirmal070125) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/entity/Test.java]: List\<String\>

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java (hunk 3 above)
Comment (yasassri) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: This is fixed.

**Pull request merged**